platform/mellanox/mlnx-platform-api/sonic_platform/utils.py

```diff
@@ -110,6 +110,44 @@ def _key_value_converter(content, delimeter):
     return ret
 
 
+def _key_value_converter_first_delim(content, delimeter):
+    ret = {}
+    for line in content.splitlines():
+        k, v = line.split(delimeter, 1)[0], line.split(delimeter, 1)[1]
+        ret[k.strip()] = v.strip()
+    return ret
+
+
+def read_key_value_file_first_delim(file_path,
```

Comment:
What is the difference from read_key_value_file?

Reply:
`read_key_value_file` separates based on a delimiter (delim = ':' string = 'abc:def' return value = ['abc','def']) which fails on multiple occurrence of the same delimiter in a line, So `read_key_value_file_first_delim`  separates based on first occurrence of delimiter(delim = ':' string = 'abc:def:cde' return value = ['abc','def:cde'])

Reply:
After discussion: Removed the new function and updated `read_key_value_file` to consider both cases